vllm/model/model.py

```diff
@@ -32,6 +35,32 @@ def __init__(self, **kwargs):
         self.vllm_base_url = None
         os.environ["HF_TOKEN"] = self.hf_secret_token
 
+    async def _monitor_vllm_health(self):
+        try:
+            if self.openai_compatible:
+                async with httpx.AsyncClient() as client:
+                    while True:
+                        response = await client.get(f"{self.vllm_base_url}/health")
+                        if response.status_code != 200:
+                            raise RuntimeError("vLLM is unhealthy")
+                        await asyncio.sleep(self.HEALTH_CHECK_INTERVAL)
+            else:
+                while True:
+                    await self.llm_engine.check_health()
+                    await asyncio.sleep(self.HEALTH_CHECK_INTERVAL)
+        except Exception as e:
+            logging.error(
+                f"vLLM has gone into an unhealthy state due to error: {e}, restarting service now..."
+            )
+            os._exit(1)
+
+    def _run_background_vllm_health_check(self):
+        logger.info("Starting background health check loop")
+        loop = asyncio.new_event_loop()
+        loop.create_task(self._monitor_vllm_health())
+        thread = threading.Thread(target=loop.run_forever, daemon=True)
+        thread.start()
+
```

Comment:
Do you think it makes sense to move these functions over to another file? It would make our model.py look a lot cleaner

Reply:
I am open to that.